packages/grid/_modules_/grid/components/toolbar/DensitySelector.tsx

```diff
@@ -88,7 +88,8 @@ export function DensitySelector() {
         startIcon={getSelectedDensityIcon()}
         onClick={handleDensitySelectorOpen}
         aria-label={apiRef!.current.getLocaleText('toolbarDensityLabel')}
-        aria-haspopup="true"
+        aria-expanded="true"
```

Comment:
It should only be `true` when the popup is open, `undefined` otherwise.

```suggestion
        aria-expanded={open ? 'true' : undefined}
```

See https://github.com/mui-org/material-ui/blob/a1e651f48deaabbe70ea1033e29e869906284b19/packages/material-ui/src/Select/SelectInput.js#L372 for an implementation of the same pattern.

It was done by @eps1lon in https://github.com/mui-org/material-ui/pull/16739 to match https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html

Reply:
Since this came up more than once recently: `aria-expanded={open ? 'true' : undefined}` instead of `aria-expanded{open}` is usually applied to implement SHOULD criteria i.e. don't get any validator warnings.

Reply:
Updated

Reply:
@eps1lon Are you saying we can simply do `aria-expanded={open}`?